src/components/app/load-data.test.js

```diff
@@ -0,0 +1,74 @@
+import { getInitialState, loadData } from './load-data';
+import { saveState } from '../../utils';
+import formatData from '../../utils/format-data';
+import loremIpsum from '../../utils/data/lorem-ipsum.mock';
+import animals from '../../utils/data/animals.mock';
+import demo from '../../utils/data/demo.mock';
+
+describe('load-data', () => {
+  describe('getInitialState', () => {
+    const loremData = loadData('lorem');
+
+    it('returns an object', () => {
+      expect(getInitialState(loremData)).toEqual(expect.any(Object));
+    });
+
+    it('does not require the second argument', () => {
+      expect(getInitialState(loremData)).toEqual(
+        getInitialState(loremData, {})
+      );
+    });
+
+    it('returns full initial state', () => {
+      expect(getInitialState(loremData)).toMatchObject({
+        ...loremData,
+        chartSize: {},
+        parameters: true,
+        textLabels: true,
+        theme: 'dark',
+        view: 'combined',
+        visible: { labelBtn: true, themeBtn: true }
+      });
+    });
+
+    it('returns mutated visible state if prop is set', () => {
+      const visible = { themeBtn: false };
+      expect(getInitialState(loremData, { visible })).toMatchObject({
+        visible: { labelBtn: true, themeBtn: false }
+      });
+    });
+
+    it('uses prop values instead of defaults if provided', () => {
+      expect(getInitialState(loremData, { theme: 'light' })).toMatchObject({
+        theme: 'light'
+      });
+    });
+
+    it('uses localstorage values instead of defaults if provided', () => {
+      const storeValues = {
+        parameters: false,
+        textLabels: false,
+        theme: 'light',
+        view: 'task'
+      };
+      saveState(storeValues);
+      expect(getInitialState(loremData)).toMatchObject(storeValues);
+    });
+
+    it('uses prop values instead of localstorage if provided', () => {
+      saveState({ theme: 'light' });
+      expect(getInitialState(loremData, { theme: 'dark' })).toMatchObject({
+        theme: 'dark'
+      });
+    });
+  });
+
+  describe('loadData', () => {
+    it('returns the correct dataset when passed a dataset string', () => {
+      expect(loadData('random')).toEqual(expect.any(Object));
```

Comment:
Would this test pass if the random data set was changed to 

```
loadData(name) => {
   switch(name) {
    case 'random':
      return {}
  }
}
```

? (what i'm asking is if we should be checking the format of random as well)

Reply:
Yeah it would. I guess I was being a bit lazy with it because I didn't consider it the right place to implement a bunch of checks on the random data generator. I can expand it though